core/src/main/java/io/smallrye/stork/utils/InMemoryAddressesBackend.java

```diff
@@ -0,0 +1,40 @@
+package io.smallrye.stork.utils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class InMemoryAddressesBackend {
```

Comment:
IMO the `final` here is about extending the class rather than threadsafety. This is a good remark, I took the old `StaticAddressesBackend` as it was and extracted to an upper level. I think we can work with a non final class.